File: div3/B.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool can_sort_array(vector<int>& arr) {
    int n = arr.size();
    bool odd_present = false;
    bool even_present = false;

    for (int num : arr) {
        if (num % 2 != 0) {
            odd_present = true;
        } else {
            even_present = true;
        }
    }

    if (!odd_present || !even_present) {
        return true; // Array contains either only odd numbers or only even numbers
    }

    // Check if we can sort the array
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] % 2 == arr[j] % 2) {
                if ((arr[i] % 2 == 0 && arr[i] > arr[j]) || (arr[i] % 2 != 0 && arr[i] < arr[j])) {
                    return false;
                }
            }
        }
    }

    return true;
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n;
        cin >> n;

        vector<int> arr(n);
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }

        cout << (can_sort_array(arr) ? "YES" : "NO") << endl;
    }

    return 0;
}

File: div3/B.py
def can_sort_array(arr):
    odd_even_at_even = 0
    even_odd_at_even = 0
    odd_even_at_odd = 0
    even_odd_at_odd = 0

    for i, num in enumerate(arr):
        if i % 2 == 0:  # Even index
            if num % 2 == 0:  # Even number
                even_odd_at_even += 1
            else:  # Odd number
                odd_even_at_even += 1
        else:  # Odd index
            if num % 2 == 0:  # Even number
                even_odd_at_odd += 1
            else:  # Odd number
                odd_even_at_odd += 1

    return odd_even_at_even == even_odd_at_even and odd_even_at_odd == even_odd_at_odd


# Read input
t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))

    # Check if the array can be sorted
    if can_sort_array(a):
        print("YES")
    else:
        print("NO")
